lib/manageiq/providers/workflows/builtin_runner.rb

```diff
@@ -12,12 +16,17 @@ def run_async!(resource, params = {}, secrets = {}, context = {})
 
           method_name = resource.sub(SCHEME_PREFIX, "")
 
-          runner_context = {"method" => method_name}
-          method_result = BuiltinMethods.public_send(method_name, params, secrets, context)
-          method_result.merge(runner_context)
-        rescue => err
-          cleanup(runner_context)
-          error!(runner_context, :cause => err.to_s)
+          begin
+            runner_context = {"method" => method_name}
+            method_result = BuiltinMethods.public_send(method_name, params, secrets, context)
+            method_result.merge(runner_context)
+          rescue NoMethodError
+            error!(runner_context, :cause => "undefined method [#{method_name}]")
```

Comment:
NOTE I added this specifically because I think that getting the method name wrong is going to be a very likely error and the payload that was returned was very much "internal ruby stuff" that would confuse a workflow author

```
{"method"=>"invalid_method",
 "running"=>false,
 "success"=>false,
 "output"=>
  {"Error"=>"States.TaskFailed",
   "Cause"=>
    "undefined method `invalid_method' for ManageIQ::Providers::Workflows::BuiltinMethods:Class\n" +
    "\n" +
    "            method_result = BuiltinMethods.public_send(method_name, params, secrets, context)\n" +
    "                                          ^^^^^^^^^^^^"}}
```